Source/ACE.Server/WorldObjects/Vendor.cs

```diff
@@ -72,12 +73,12 @@ public override void ActOnUse(WorldObject wo)
 
             var rotateTime = Rotate(player);    // vendor rotates towards player
 
-            // TODO: remove this when DelayManager is not forward propagating current tick time
+            var delayUntilTime = Time.GetFutureUnixTime(rotateTime);
 
             var actionChain = new ActionChain();
-            actionChain.AddDelaySeconds(0.001f);  // force to run after rotate.EnqueueBroadcastAction
+            actionChain.AddDelayForOneTick();  // force to run after rotate.EnqueueBroadcastAction
             actionChain.AddAction(this, LoadInventory);
-            actionChain.AddDelaySeconds(rotateTime);
+            actionChain.AddDelayUntil(delayUntilTime);
```

Comment:
AddDelayUntil is also a function that is not needed, unless we just do a mass search/replace for every call to 'AddDelaySeconds' and change everything to 'AddDelayUntil'

The game server does not need to build action chains with time based on the amount of time the previous actions in the chain took to execute. In fact, this could be considered an unsafe design. If code wants to be designed this way (and I can't think of 1 place in the code that does), it should be chained for the first function to build the next action